bin/migration/tables/cases.py

```diff
@@ -0,0 +1,101 @@
+from .helpers import check_existing_record, parse_to_timestamp, audit_entry_creation, log_failed_imports
+import uuid
+
+class CaseManager:
+    def __init__(self, source_cursor):
+        self.source_cursor = source_cursor
+        self.failed_imports = set()
+
+    def get_data(self):
+        self.source_cursor.execute("SELECT * FROM public.cases")
+        return self.source_cursor.fetchall()
+
+    def migrate_data(self, destination_cursor, source_data):
+        # creating a temporary table for the unique case_ref values
+        destination_cursor.execute(
+            """CREATE TABLE IF NOT EXISTS public.temp_cases (
+                                    booking_id UUID PRIMARY KEY,
+                                    case_id UUID,
+                                    reference VARCHAR(25),
+                                    court_name VARCHAR(250),
+                                    court_id UUID,
+                                    created_at VARCHAR(50),
+                                    created_by VARCHAR(100),
+                                    modified_at VARCHAR(50)
+                                    )
+                                """
+        )
+
+        destination_cursor.execute("SELECT id FROM public.courts WHERE name = 'Default Court'")
+        default_court_id = destination_cursor.fetchone()
+
+        temp_cases_data = []
+        for case in source_data:
+            reference = case[1]
+            
+            if reference is None:
+                self.failed_imports.add(('cases', case[0], 'no case reference'))
+                continue
+
+            if not check_existing_record(destination_cursor,'temp_cases', 'reference', reference):
+                destination_cursor.execute(
+                    "SELECT id FROM public.courts WHERE name = %s", (case[2],)
+                )
+                court_id = destination_cursor.fetchone()
+                booking_id= case[0]
+                case_id = str(uuid.uuid4())
+                court_name = case[2]
+                created_at = parse_to_timestamp(case[5])
+                created_by = case[4]
+                modified_at = parse_to_timestamp(case[6])
+                temp_cases_data.append((booking_id, case_id, reference, court_name, court_id, created_at, created_by, modified_at))
+
+        destination_cursor.executemany(
+            """INSERT INTO public.temp_cases (booking_id, case_id, reference, court_name, court_id, created_at, created_by, modified_at)
```

Comment:
@mazzopardi2  'created_at' and 'modified_at' fields are not populated with the actual created and modified date and time. It is displaying the migrated time.

Reply:
@JeyaSolirius - good point. There was a different time format that wasn't being handled in the parse_to_timestamp helper function. This has been fixed in the last commit